### PR TITLE
Feat(dubbo): allow using variable inside dubbo_pass

### PR DESCRIPTION
**Pull request by spacewander**
Now we can configure `dubbo_pass` like:

dubbo_pass $dubbo_service_name $dubbo_service_version $dubbo_method dubbo_backend;

### CODE REVIEW

**Comment by MengqiWu** [comment]
@Weiliang-Li

File: modules/mod_dubbo/ngx_http_dubbo_module.c [hunk 1]
```diff
@@ -649,6 +653,39 @@ ngx_http_dubbo_create_dubbo_request(ngx_http_request_t *r, ngx_connection_t *pc,
...
     dlcf = ngx_http_get_module_loc_conf(r, ngx_http_dubbo_module);
 
+    service_name = ngx_palloc(r->pool, sizeof(ngx_str_t));
+    if (service_name == NULL) {
+        return NGX_ERROR;
+    }
+
+    if (ngx_http_complex_value(r, &dlcf->service_name, service_name)
+        != NGX_OK)
+    {
```
Comment (MengqiWu) [inline modules/mod_dubbo/ngx_http_dubbo_module.c:663]: code style on '{'  and some other

**Comment by MengqiWu** [comment]
document and example need update？

File: modules/mod_dubbo/ngx_http_dubbo_module.c (hunk 1 above)
Comment (spacewander) [reply modules/mod_dubbo/ngx_http_dubbo_module.c:663]: Look like Nginx also put the `{` into a new line in similar case?
https://github.com/nginx/nginx/blob/402f4cda0c389242a587eff7bb55959663504c8b/src/http/modules/ngx_http_gzip_static_module.c#L158

Comment (MengqiWu) [reply modules/mod_dubbo/ngx_http_dubbo_module.c:663]: got it

**Comment by spacewander** [comment]
Document changed.

**Comment by MengqiWu** [comment]
+1 and wait @Weiliang-Li

**Comment by runninghai** [comment]
+1

**Comment by MengqiWu** [comment]
@runninghai

**Pull request merged**